contracts/RiskSteward/MarketCapsRiskSteward.sol

```diff
@@ -0,0 +1,168 @@
+     */
+    error InvalidMaxIncreaseBps();
+
+    /**
+     * @notice Thrown when an updateType that is not supported is operated on
+     */
+    error UnsupportedUpdateType();
+
+    /**
+     * @notice Thrown when the new value of an update is our of range
+     */
+    error UpdateNotInRange();
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(address corePoolComptroller_) {
+        require(corePoolComptroller_ != address(0), "Core Pool Comptroller address must not be zero");
+        CORE_POOL_COMPTROLLER = ICorePoolComptroller(corePoolComptroller_);
+        _disableInitializers();
+    }
+
+    function initialize(address accessControlManager_, uint256 maxIncreaseBps_) external initializer {
+        __Ownable2Step_init();
+        __AccessControlled_init_unchained(accessControlManager_);
+        if (maxIncreaseBps_ == 0) {
+            revert InvalidMaxIncreaseBps();
+        }
+        maxIncreaseBps = maxIncreaseBps_;
+    }
+
+    function setMaxIncreaseBps(uint256 maxIncreaseBps_) external {
+        _checkAccessAllowed("setMaxIncreaseBps(uint256)");
+        if (maxIncreaseBps_ == 0) {
+            revert InvalidMaxIncreaseBps();
+        }
+        maxIncreaseBps = maxIncreaseBps_;
+    }
+
+    /**
+     * @notice
+     * @param update
+     */
+    function processUpdate(RiskParameterUpdate calldata update) external {
+        _checkAccessAllowed("processUpdate(RiskParameterUpdate)");
+        if (Strings.equal(update.updateType, "MarketSupplyCaps")) {
+            _processSupplyCapUpdate(update);
+        } else if (Strings.equal(update.updateType, "MarketBorrowCaps")) {
+            _processBorrowCapUpdate(update);
+        } else {
+            revert UnsupportedUpdateType();
+        }
+    }
+
+    function _updateSupplyCaps(address market, bytes memory newValue) internal {
+        address comptroller = IVToken(market).comptroller();
+        address[] memory newSupplyCapMarkets = new address[](1);
+        newSupplyCapMarkets[0] = market;
+        uint256[] memory newSupplyCaps = new uint256[](1);
+
+        newSupplyCaps[0] = uint256(bytes32(newValue));
+        if (comptroller == address(CORE_POOL_COMPTROLLER)) {
+            ICorePoolComptroller(comptroller)._setMarketSupplyCaps(newSupplyCapMarkets, newSupplyCaps);
+        } else {
+            IIsolatedPoolsComptroller(comptroller).setMarketSupplyCaps(newSupplyCapMarkets, newSupplyCaps);
+        }
+    }
+
+    function _updateBorrowCaps(address market, bytes memory newValue) internal {
+        address comptroller = IVToken(market).comptroller();
+        address[] memory newBorrowCapMarkets = new address[](1);
+        newBorrowCapMarkets[0] = market;
+        uint256[] memory newBorrowCaps = new uint256[](1);
+        newBorrowCaps[0] = uint256(bytes32(newValue));
+        if (comptroller == address(CORE_POOL_COMPTROLLER)) {
+            ICorePoolComptroller(comptroller)._setMarketBorrowCaps(newBorrowCapMarkets, newBorrowCaps);
```

Comment:
Might be a good time to get rid of these dreaded underscores in the core pool comptroller, or at least add an alias. I know this would require an audit, and the diamond upgrade, and all the pain associated with it, but we have to work towards unifying the interfaces at some point.